components/micro-gateway-core/src/main/ballerina/src/gateway/utils/api_key_utils.bal

```diff
@@ -187,3 +167,27 @@ public function getAPIKeyAuth() returns [string, string] | error {
         return error("API key is missing in invocation context");
     }
 }
+
+public function getAPIsforAPIKey() returns json[] {
+    json[] apis = [];
+    map<anydata>[] | error apiList = map<anydata>[].constructFrom(config:getAsArray(API_KEY_ISSUER_API));
+    if (apiList is map<anydata>[] && apiList.length() > 0) {
+        foreach map<anydata> apiData in apiList {
+            anydata apiName = apiData[NAME];
+            anydata apiVersions = apiData[VERSIONS];
+            if (apiName is string) {
+                if (!(apiVersions is string) || (stringutils:equalsIgnoreCase("*", apiVersions))) {
```

Comment:
Shall we use simple equals here? No need of `equalsIgnoreCase` when comparing a special `char`?
```suggestion
                if (!(apiVersions is string) || "*" == apiVersions) {
```

Reply:
done